src/main/java/uk/gov/hmcts/ccd/domain/service/callbacks/CallbackService.java

```diff
@@ -97,23 +110,42 @@ public CallbackService(final SecurityUtils securityUtils,
     }
 
     private <T> Optional<ResponseEntity<T>> sendRequest(final String url,
+                                                        final String callbackType,
                                                         final Class<T> clazz,
+
                                                         final CallbackRequest callbackRequest) {
+        Instant startTime = null;
+        int httpStatus = 0;
         try {
             LOG.debug("Invoking callback {}", url);
             final HttpHeaders httpHeaders = new HttpHeaders();
             httpHeaders.add("Content-Type", "application/json");
+
             final HttpHeaders securityHeaders = securityUtils.authorizationHeaders();
             if (null != securityHeaders) {
                 securityHeaders.forEach((key, values) -> httpHeaders.put(key, values));
             }
             final HttpEntity requestEntity = new HttpEntity(callbackRequest, httpHeaders);
-            return Optional.ofNullable(restTemplate.exchange(url, HttpMethod.POST, requestEntity, clazz));
+
+            startTime = Instant.now();
```

Comment:
shouldn't this be initialized at L 117?

Reply:
What if S2S token generation itself fails? finally block will use a null startTime.
How about doing all time consuming activity on top of the method and after that initialising the startTime?

Reply:
Yeah.. good point. Lets move s2s bit on top of the method.